Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,320 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.NormNum.Prime
+import Mathlib.Algebra.GeomSum
+import Mathlib.Tactic.FinCases
+
+/-!
+## FactorisationProperties.lean
+This file defines abundant, semi-perfect, deficient, and weird numbers and formalizes their
+  relations with prime and perfect numbers.
+
+## Main Definitions
+Let `n : ℕ`. All the following definitions are in the Nat namespace.
+* `Abundant` a natural number `n` is abundant if the sum of its proper divisors are greater than
+  itself and it is greater than zero
+* `Pseudoperfect` a natural number `n` is semi-perfect if a subset of its proper divisors equals
+  itself and it is greater than zero
+* `Deficient` a natural number `n` is deficient if the sum of its proper divisors is less than
+  itself and it is greater than zero
+* `WeirdNumber` a natural number is weird if it is both abundant and not semi-perfect
+
+## Main Results
+* `divisors_eq_proper_union_self` the `Finset` that is `Nat.divisors` is equal to the union of
+  `Nat.properDivisors` and the number itself
+* `deficient_or_perfect_or_abundant` any natural number greater than zero is either deficient,
+  perfect, or abundant
+* `prime_deficient` all prime natural numbers are deficient
+* `prime_pow_deficient` any power of a prime number is deficient
+
+## Implementation Details
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+abundant deficient weird Pseudoperfect
+-/
+
+open Nat BigOperators Finset
+
+set_option maxRecDepth 1000000
```

Comment:
In general, this line makes me suspicious: is something odd going on? Was this line needed at some point and is not any more?

Reply:
Sorry I forgot to bring this up in my original comment. Yes this is necessary for proving that 70 is a weird number because fin_cases checks every iteration of the sum of each subset of the proper divisors of 70 to prove that none of them equal 70. Is there a way to centralize it only to that proof. I believe David had an issue with this initially because it could be dangerous.